bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BValueArray.java

```diff
@@ -0,0 +1,573 @@
+        return byteValues[(int) index];
+    }
+
+    public double getFloat(long index) {
+        rangeCheckForGet(index, size);
+        return floatValues[(int) index];
+    }
+
+    public String getString(long index) {
+        rangeCheckForGet(index, size);
+        return stringValues[(int) index];
+    }
+
+    // ----------------------------  add methods --------------------------------------------------
+
+    public void add(long index, BRefType<?> value) {
+        synchronized (this) {
+            if (freezeStatus.getState() != CPU.FreezeStatus.State.UNFROZEN) {
+                handleInvalidUpdate(freezeStatus.getState());
+            }
+        }
+
+        prepareForAdd(index, refValues.length);
+        refValues[(int) index] = value;
+    }
+
+    public void add(long index, long value) {
+        synchronized (this) {
+            if (freezeStatus.getState() != CPU.FreezeStatus.State.UNFROZEN) {
+                handleInvalidUpdate(freezeStatus.getState());
+            }
+        }
+
+        prepareForAdd(index, intValues.length);
+        intValues[(int) index] = value;
+    }
+
+    public void add(long index, int value) {
+        if (elementType.getTag() == TypeTags.INT_TAG) {
+            add(index, (long) value);
+            return;
+        }
+
+        synchronized (this) {
+            if (freezeStatus.getState() != CPU.FreezeStatus.State.UNFROZEN) {
+                handleInvalidUpdate(freezeStatus.getState());
+            }
+        }
+
+        prepareForAdd(index, booleanValues.length);
+        booleanValues[(int) index] = value;
+    }
+
+    public void add(long index, byte value) {
+        synchronized (this) {
+            if (freezeStatus.getState() != CPU.FreezeStatus.State.UNFROZEN) {
+                handleInvalidUpdate(freezeStatus.getState());
+            }
+        }
+
+        prepareForAdd(index, byteValues.length);
+        byteValues[(int) index] = value;
+    }
+
+    public void add(long index, double value) {
+        synchronized (this) {
+            if (freezeStatus.getState() != CPU.FreezeStatus.State.UNFROZEN) {
+                handleInvalidUpdate(freezeStatus.getState());
+            }
+        }
+
+        prepareForAdd(index, floatValues.length);
+        floatValues[(int) index] = value;
+    }
+
+    public void add(long index, String value) {
+        synchronized (this) {
+            if (freezeStatus.getState() != CPU.FreezeStatus.State.UNFROZEN) {
+                handleInvalidUpdate(freezeStatus.getState());
+            }
+        }
+
+        prepareForAdd(index, stringValues.length);
+        stringValues[(int) index] = value;
+    }
+
+    //-------------------------------------------------------------------------------------------------------------
+
+    public void append(BRefType<?> value) {
+        add(size, value);
+    }
+
+    @Override
+    public BType getType() {
+        return arrayType;
+    }
+
+    @Override
+    public void stamp(BType type) {
```

Comment:
in cases like stamping an int[] to a any[], shouldn't we move the values from intValues to refValues, and vice versa?

Reply:
Ya, we have to implement that. Still pending.

Reply:
Done.